core/services/feeds/service.go

```diff
@@ -503,6 +539,42 @@ func (s *service) CancelJobProposal(ctx context.Context, id int64) error {
 	return err
 }
 
+// ListSpecsByJobProposalIDs gets the specs which belong to the job proposal ids.
+func (s *service) ListSpecsByJobProposalIDs(ids []int64) ([]JobProposalSpec, error) {
+	return s.orm.ListSpecsByJobProposalIDs(ids)
+}
+
+// GetSpec gets the spec details by id.
+func (s *service) GetSpec(id int64) (*JobProposalSpec, error) {
+	return s.orm.GetSpec(id)
+}
+
+// UpdateSpecDefinition updates the spec's TOML definition.
+func (s *service) UpdateSpecDefinition(ctx context.Context, id int64, defn string) error {
+	pctx := pg.WithParentCtx(ctx)
+
+	spec, err := s.orm.GetSpec(id, pctx)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return errors.Wrap(err, "job proposal spec does not exist")
```

Comment:
minor but maybe better for it to be .New instead of .Wrap?

Reply:
We want to wrap so we can detect the presence of an ErrNoRows in a `errors.Is` check